cmd/docker/docker.go

```diff
@@ -168,12 +169,29 @@ func hideUnsupportedFeatures(cmd *cobra.Command, clientVersion string, hasExperi
 func isSupported(cmd *cobra.Command, clientVersion string, hasExperimental bool) error {
 	if !hasExperimental {
 		if _, ok := cmd.Tags["experimental"]; ok {
-			return errors.New("only supported with experimental daemon")
+			return errors.New("only supported on a Docker daemon with experimental features enabled")
 		}
 	}
 
 	if cmdVersion, ok := cmd.Tags["version"]; ok && versions.LessThan(clientVersion, cmdVersion) {
-		return fmt.Errorf("only supported with daemon version >= %s", cmdVersion)
+		return fmt.Errorf("requires API version %s, but the Docker daemon is version %s", cmdVersion, clientVersion)
+	}
+
+	errs := []string{}
+
+	cmd.Flags().VisitAll(func(f *pflag.Flag) {
+		if f.Changed {
+			if flagVersion, ok := f.Annotations["version"]; ok && len(flagVersion) == 1 && versions.LessThan(clientVersion, flagVersion[0]) {
```

Comment:
This line is rather long, and duplicated in `hideUnsupportedFeatures()`. Could you move it out into a new function:

```go
func flagIsNotSupported(f *pflag.Flag, clientVersion string) { ... }
```